packages/common-scripts/src/common.ts

```diff
@@ -590,26 +589,129 @@ function _deductCapacity(
   return [txSkeleton, capacity];
 }
 
-export async function setupInputCell(
+// Alter output generated by `setupInputCell`
+async function collectInput(
   txSkeleton: TransactionSkeletonType,
   inputCell: Cell,
-  fromInfo: FromInfo,
+  fromInfo?: FromInfo,
   {
     config = undefined,
-    needCapacity = undefined,
     since = undefined,
-    defaultWitness = undefined,
+    defaultWitness = "0x",
+    needCapacity = undefined,
   }: Options & {
-    needCapacity?: HexString;
-    since?: PackedSince;
     defaultWitness?: HexString;
+    since?: PackedSince;
+    needCapacity?: bigint;
   } = {}
 ): Promise<{
   txSkeleton: TransactionSkeletonType;
-  availableCapacity: HexString;
+  availableCapacity: bigint;
 }> {
   config = config || getConfig();
 
+  txSkeleton = await setupInputCell(txSkeleton, inputCell, fromInfo, {
+    config,
+    since,
+    defaultWitness,
+  });
+
+  const lastOutputIndex: number = txSkeleton.get("outputs").size - 1;
+  const lastOutput: Cell = txSkeleton.get("outputs").get(lastOutputIndex)!;
+  const lastOutputCapacity: bigint = BigInt(lastOutput.cell_output.capacity);
+  const lastOutputFixedEntryIndex: number = txSkeleton
+    .get("fixedEntries")
+    .findIndex((fixedEntry) => {
+      return (
+        fixedEntry.field === "outputs" && fixedEntry.index === lastOutputIndex
+      );
+    });
+  const fromScript: Script = inputCell.cell_output.lock;
+
+  let availableCapacity: bigint = 0n;
+  // TODO: remove this special treatment.
+  if (isAcpScript(fromScript, config)) {
+    const destroyable: boolean = !!(
+      fromInfo &&
+      typeof fromInfo === "object" &&
+      "destroyable" in fromInfo &&
+      fromInfo.destroyable
+    );
+    needCapacity = needCapacity || lastOutputCapacity;
+
+    if (destroyable) {
+      availableCapacity = lastOutputCapacity;
+      // remove output & fixedEntry added by `setupInputCell`
+      txSkeleton = txSkeleton.update("outputs", (outputs) => {
+        return outputs.remove(lastOutputIndex);
+      });
+      if (lastOutputFixedEntryIndex >= 0) {
+        txSkeleton = txSkeleton.update("fixedEntries", (fixedEntries) => {
+          return fixedEntries.remove(lastOutputFixedEntryIndex);
+        });
+      }
+    } else {
+      // FIXME: fixedEntries ignored.
```

Comment:
And here, internal to a function, it should be fine if we ignore fixedEntries setting for cells that are just added.

Reply:
Removed.